src/test/java/bio/terra/app/controller/SnapshotsApiControllerTest.java

```diff
@@ -261,6 +279,181 @@ void unlockSnapshot() throws Exception {
     verify(snapshotService).manualExclusiveUnlock(TEST_USER, SNAPSHOT_ID, unlockRequest);
   }
 
+  @Test
+  void createSnapshot() throws Exception {
+    mockValidators();
+
+    SnapshotRequestModel requestModel =
+        jsonLoader.loadObject("ingest-test-snapshot.json", SnapshotRequestModel.class);
+    when(snapshotService.getSourceDatasetIdsFromSnapshotRequest(requestModel))
+        .thenReturn(List.of(DATASET_ID));
+
+    IamAction iamAction = IamAction.LINK_SNAPSHOT;
+    when(iamService.isAuthorized(
+            TEST_USER, IamResourceType.DATASET, DATASET_ID.toString(), iamAction))
```

Comment:
I disagree: thinking of it from a CRUD perspective, snapshot creation taking place within the snapshots API makes sense to me and aligns with the organization we see in other Terra services (ex. Rawls [workspace creation](https://rawls.dsde-prod.broadinstitute.org/#/workspaces/createWorkspace) is a workspaces endpoint rather than a billing endpoint).